cmd/insights/contributors.go

```diff
@@ -0,0 +1,289 @@
+package insights
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"sync"
+
+	"github.com/charmbracelet/bubbles/table"
+	"github.com/charmbracelet/lipgloss"
+	"github.com/open-sauced/go-api/client"
+	"github.com/open-sauced/pizza-cli/cmd/common"
+	repoquery "github.com/open-sauced/pizza-cli/cmd/repo-query"
+	"github.com/open-sauced/pizza-cli/pkg/api"
+	"github.com/spf13/cobra"
+)
+
+type Options struct {
+	// APIClient is the http client for making calls to the open-sauced api
+	APIClient *client.APIClient
+	// Repos is the array of git repository urls
+	Repos []string
+	// File is the path to yaml file containing an array of git repository urls
+	File string
+	// Period is the number of days, used for query filtering
+	Period int32
+}
+
+// NewContributorsCommand returns a new cobra command for 'pizza insights contributors'
+func NewContributorsCommand() *cobra.Command {
+	opts := &Options{}
+	cmd := &cobra.Command{
+		Use:   "contributors [flags]",
+		Short: "Gather insights about contributors of indexed git repositories",
+		Long:  "Gather insights about contributors of indexed git repositories. This command will show new, recent, churn, repeat contributors for each git repository",
+		Args: func(cmd *cobra.Command, args []string) error {
+			fileFlag := cmd.Flags().Lookup(common.FlagNameFile)
+			reposFlag := cmd.Flags().Lookup(common.FlagNameRepos)
+			if !fileFlag.Changed && !reposFlag.Changed {
+				return fmt.Errorf("must pass either %s or %s flag", fileFlag.Name, reposFlag.Name)
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx, apiClient, err := api.DefaultClientSetup(cmd)
+			if err != nil {
+				return err
+			}
+			opts.APIClient = apiClient
+			return run(ctx, opts)
+		},
+	}
+	cmd.Flags().StringVarP(&opts.File, common.FlagNameFile, "f", "", "Path to yaml file containing an array of git repository urls")
+	cmd.Flags().StringSliceVarP(&opts.Repos, common.FlagNameRepos, "r", nil, "Comma-separated array of git repository urls")
+	cmd.Flags().Int32VarP(&opts.Period, common.FlagNamePeriod, "p", 30, "Number of days, used for query filtering")
+	cmd.MarkFlagsMutuallyExclusive(common.FlagNameFile, common.FlagNameRepos)
+	return cmd
+}
+
+func run(ctx context.Context, opts *Options) error {
+	repositories, err := common.HandleRepositoryValues(opts.Repos, opts.File)
+	if err != nil {
+		return err
+	}
+	var (
+		errorChan = make(chan error)
+		doneChan  = make(chan bool)
+		waitGroup = new(sync.WaitGroup)
+	)
+	for repoURL := range repositories {
+		waitGroup.Add(1)
+		go func(repoURL string) {
+			defer waitGroup.Done()
+			repoContributorInsights, err := findAllRepositoryContributors(ctx, opts, repoURL)
+			if err != nil {
+				errorChan <- err
+				return
+			}
+			if repoContributorInsights == nil {
+				return
+			}
+			repoContributorInsights.RenderTable()
+		}(repoURL)
+	}
+	go func() {
+		waitGroup.Wait()
+		close(doneChan)
+	}()
```

Comment:
This looks like a similar structuring as the bake command refactor you have: I don't think having the `waitgroup.Wait()` in a separate thread make sense. The main through should just wait for these all to finish.

And similar comments to the `errorChan`: these will get squashed and only the first error will surface and be returned.